File: AST/SingleIf.java
package AST;

import AST.Visitor.Visitor;
import java_cup.runtime.ComplexSymbolFactory.Location;

public class SingleIf extends If {

  public SingleIf(Exp ae, Statement s, Location pos) {
    super(pos);
    cond=ae; 
    s1=s;
  }

  public void accept(Visitor v) {
    v.visit(this);
  }
}

File: AST/SingleIf.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AST
{
    public class SingleIf : AstNode, Statement
    {
        public Exp Cond;
        public List<AstNode> Statements;

        public SingleIf(Exp cond,  List<AstNode> statements, int lineNumber, int startCol)
        {
            Cond = cond;
            Statements = statements;
            LineNumber = lineNumber;
            StartCol = startCol;
        }

        public override void Accept(Visitor v) => v.Visit(this);
    }
}
